test/rules/avoid_positional_boolean_parameters.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N avoid_positional_boolean_parameters`
+
+void good({bool a}) {} // OK
+
+void bad(bool a) {} // LINT
+
+class A {
+  void good({bool a}) {} // OK
+
+  void bad(bool a) {} // LINT
+}
+
+class B {
+  static void good({bool a}) {} // OK
+
+  static void bad(bool a) {} // LINT
+}
+
+class C {
+  bool value;
+  C.good({bool value}) { // OK
+    this.value = value;
+  }
+
+  C.bad(bool a) { // LINT
+    this.value = value;
+  }
+}
+
+class D {
+  bool value;
+  D.good({this.value}); // OK
+
+  D.bad(this.value); // LINT
+
+}
```

Comment:
There should be at least one test that uses optional positional parameters
```
void bad([bool value]) {}
```
and at least one test with a setter
```
void set good(bool value) {}
```